src/java.base/share/classes/java/lang/template/StringTemplate.java

```diff
@@ -0,0 +1,368 @@
+     * {@code values} list size.
+     *
+     * @param fragments list of string fragments
+     * @param values    list of expression values
+     *
+     * @return StringTemplate composed from string
+     *
+     * @throws IllegalArgumentException if fragments list size is not one more
+     *         than values list size
+     * @throws NullPointerException if fragments is null or values is null or if any fragment is null.
+     *
+     * @implNote Contents of both lists are copied to construct immutable lists.
+     */
+    public static StringTemplate of(List<String> fragments, List<Object> values) {
+        Objects.requireNonNull(fragments, "fragments must not be null");
+        Objects.requireNonNull(values, "values must not be null");
+        if (values.size() + 1 != fragments.size()) {
+            throw new IllegalArgumentException(
+                    "fragments list size is not one more than values list size");
+        }
+        for (String fragment : fragments) {
+            Objects.requireNonNull(fragment, "fragments elements must be non-null");
+        }
+        fragments = Collections.unmodifiableList(new ArrayList<>(fragments));
+        values = Collections.unmodifiableList(new ArrayList<>(values));
+        return new SimpleStringTemplate(fragments, values);
+    }
+
+     /**
+     * Creates a string that interleaves the elements of values between the
+     * elements of fragments.
+     *
+     * @param fragments  list of String fragments
+     * @param values     list of expression values
+     *
+     * @return String interpolation of fragments and values
+     *
+     * @throws NullPointerException fragments or values is null or if any of the fragments is null
+     */
+    public static String interpolate(List<String> fragments, List<Object> values) {
+        return TemplateRuntime.interpolate(fragments, values);
+    }
+
+    /**
+      * Combine one or more {@link StringTemplate StringTemplates} to produce a combined {@link StringTemplate}.
+      * {@snippet :
+      * StringTemplate st = StringTemplate.combine(RAW."\{a}", RAW."\{b}", RAW."\{c}");
+      * assert st.interpolate().equals(RAW."\{a}\{b}\{c}");
+      * }
+      *
+      * @param sts  one or more {@link StringTemplate}
+      *
+      * @return combined {@link StringTemplate}
+      *
+      * @throws NullPointerException if sts is null or if any of the elements are null
+      * @throws RuntimeException if sts has zero elements
+      */
+    public static StringTemplate combine(StringTemplate... sts) {
+        return TemplateRuntime.combine(sts);
+    }
+
+    /**
+     * Interpolation template processor instance.
+     * Example: {@snippet :
+     * int x = 10;
+     * int y = 20;
+     * String result = STR."\{x} + \{y} = \{x + y}"; // @highlight substring="STR"
+     * }
+     * @implNote The result of interpolation is not interned.
+     */
+    public static final StringProcessor STR = st -> st.interpolate();
```

Comment:
Should be `StringTemplate::interpolate`.

Reply:
Yep